### PR TITLE
add ZIR compare output test case to test suite

### PR DESCRIPTION
**Pull request by andrewrk**

### CODE REVIEW

**Comment by andrewrk** [comment]
@tiehuis I would be interested in feedback on 5e8934e6d4c9eb5958420328b24884261aa7fca0

**Comment by andrewrk** [comment]
I'm planning to merge this. I'm pretty sure the changes to big ints are good long term changes. The API takes a bit more initial learning, but ultimately is more flexible, which is needed by the self-hosted compiler.

**Comment by tiehuis** [comment]
I generally like the design and pushing the checks from runtime to compile-time from the quick skim I've had. I'll try set some time aside for a closer look but looks good on a first glance!

File: lib/std/child_process.zig [hunk 1]
```diff
@@ -46,6 +46,12 @@ pub const ChildProcess = struct {
 
     /// Set to change the current working directory when spawning the child process.
     cwd: ?[]const u8,
+    /// Set to change the current working directory when spawning the child process.
+    /// This is not yet implemented for Windows. See https://github.com/ziglang/zig/issues/5190
+    /// Once that is done, `cwd` will be deprecated in favor of this field.
+    /// The directory handle must be opened with the ability to be passed
+    /// to a child process (no `O_CLOEXEC` flag on POSIX).
```
Comment (daurnimator) [inline lib/std/child_process.zig:53]: This will cause races: you should always remove the `O_CLOEXEC` in the forked process (before the exec)

Comment (andrewrk) [reply lib/std/child_process.zig:53]: Can you explain or link to reading material?

Comment (daurnimator) [reply lib/std/child_process.zig:53]: Take a situation where 2 threads simultaneously spawn a child process: you don't want one messing with fds to affect the other ==> you need to do all the fd manipulation in the child (after fork; before exec). `posix_spawn` was invented to solve this problem (as its very hard to do correctly with respect to signals).

Comment (andrewrk) [reply lib/std/child_process.zig:53]: Ahh that makes sense, thanks! Would you mind making this note on #5190 ?

**Pull request merged**